### PR TITLE
Parsedown is GFM

### PR DESCRIPTION
**Pull request by aidantwoods**
https://github.com/erusev/parsedown#features :)

### CODE REVIEW

File: src/Markbench/Driver/ParsedownDriver.php [hunk 1]
```diff
@@ -44,7 +44,7 @@ public function getName()
      */
     public function getDialect()
     {
-        return null;
+        return 'gfm';
```
Comment (cebe) [inline src/Markbench/Driver/ParsedownDriver.php:47]: :+1:

File: src/Markbench/Driver/ParsedownExtraDriver.php [hunk 2]
```diff
@@ -44,7 +44,7 @@ public function getName()
      */
     public function getDialect()
     {
-        return 'extra';
+        return 'gfm & extra';
```
Comment (cebe) [inline src/Markbench/Driver/ParsedownExtraDriver.php:47]: isn't extra a superset of gfm? I mean if you look at it in very detail there are probably things that do not match, but both have no detailed spec so I'd leave `extra` here.

Comment (aidantwoods) [reply src/Markbench/Driver/ParsedownExtraDriver.php:47]: GFM is a strict superset of CommonMark: https://github.github.com/gfm/

Extra explicitly conflicts with GFM when doing HTML block parsing :)

Comment (cebe) [reply src/Markbench/Driver/ParsedownExtraDriver.php:47]: Oh well, a lot has changed since I looked at these in detail :)

**Comment by cebe** [comment]
Thank you!

File: src/Markbench/Driver/ParsedownExtraDriver.php (hunk 2 above)
Comment (aidantwoods) [reply src/Markbench/Driver/ParsedownExtraDriver.php:47]: GFM used to be as about underspecified as Gruber Markdown, so you're not wrong to think it didn't have a detailed spec at one point :)

GitHub ended up writing it more formally as a CommonMark extension at some point, but kept the same name (I remember a couple years ago the GitHub parser was horrible and kept breaking my comments when they changed things slightly! :p).

Comment (cebe) [reply src/Markbench/Driver/ParsedownExtraDriver.php:47]: btw, I have just seen https://github.com/Parsemd/Parsemd-PHP is there something interesting coming out of that or was it just an experiment?

Comment (aidantwoods) [reply src/Markbench/Driver/ParsedownExtraDriver.php:47]: There might be something coming from it, probably not in its current form though (although if I recall I implemented quite a bit of the spec, so it's probably pretty close to usable).

The overall design started disintegrating when trying to resolve some places where it deviated from spec. I could probably keep going, but it seemed like there were too many cases of special handling popping up that it wasn't going to be a sure thing that arbitrary dialects would work as hoped.

I ended up restarting (and redesigning) the project in Swift where I can lean on the type system a little more with the intent of porting it back over to PHP, I haven't really touched it in a while though – might start working on it again soon.

https://github.com/Parsemd/Parsemd-PHP was still a useful venture – at least I'm now aware of which problems to bare in mind when designing the new architecture :p

Comment (cebe) [reply src/Markbench/Driver/ParsedownExtraDriver.php:47]: cool, thanks for the detailed response :)

Comment (aidantwoods) [reply src/Markbench/Driver/ParsedownExtraDriver.php:47]: No worries :)

Something I ended up being quite pleased with is the abstract emphasis implementation passed all the CommonMark emphasis examples (and I'm pretty sure also agrees completely with spec) when specialised to the CommonMark specifics (marker run length, markers etc..)

**Pull request merged**